checker/errors/errors.go

```diff
@@ -23,20 +23,27 @@ type TODO struct {
 	filename string
 	lines    []string
 	linecnt  int
+	issueId  string
+}
+
+type metadata struct {
+	IssueId string `json:"issueID"`
 }
 
 // ToJSON converts the todo error into json format
 func (err *TODO) ToJSON() ([]byte, error) {
 	res := &struct {
-		Type     string `json:"type"`
-		Filename string `json:"filename"`
-		Line     int    `json:"line"`
-		Message  string `json:"message"`
+		Type     string   `json:"type"`
+		Filename string   `json:"filename"`
+		Line     int      `json:"line"`
+		Message  string   `json:"message"`
+		Metadata metadata `json:"metadata"`
```

Comment:
The metadata should be a `map[string]string` instead of a typed object. The reason is that different errors might have different metadata in them.

For example, `issueID` is a field applicable to non existent issue error but not for todo malformed error

Reply:
Changed to `map[string]string`